relayer/relayer.go

```diff
@@ -105,9 +107,19 @@ func NewRelayer(
 		logger:                   logger,
 	}
 
-	// Start the message router. We must do this before Subscribing for the first time, otherwise we may miss an incoming message
+	// Start the message router. We must do this before Subscribing or Initializing for the first time, otherwise we may miss an incoming message
 	go r.RouteToMessageChannel()
 
+	// Initialize the subscriber. This will poll the node for any logs that match the filter query from the stored block height,
+	// and process the contained warp messages. If initialization fails, continue with normal relayer operation, but log the error.
+	err = sub.Initialize()
```

Comment:
I think there is a possibility to still not process messages from all blocks here because the subscription is opened until after we finish `Initialize`, so there could theoretically be new blocks produced/accepted in between those two events. 

I'm not sure how complicated the implementation would be, but I think the desired behavior would be:
1. Open the subscription to be notified about logs in new blocks.
2. Get the current block height on chain.
3. Process messages from any blocks from the "latest processed" height up to the current block height.
4. Start processing any messages received over the subscription.

Reply:
Yup, I think that should be fine, as long as we grab the latest processed block from the db before opening the subscription. Message order wouldn't be guaranteed, but that's not part of the relayer spec anyway.

Reply:
Updated now.

Reply:
RIght now I think we're running `subscribe` and `processFromHeight` in two separate go routines concurrently, so if new blocks came before `processFromHeight` gets back the latest block height, we'd double process that block right?

Reply:
That's correct. I think it would be somewhat tricky to cover all the edge cases here, especially since the `eth_subscribe` and `eth_ blockNumber` calls cannot be done atomically. The alternative I'm currently in favor of is implementing https://github.com/ava-labs/awm-relayer/issues/22 to guard against double processing, rather than doing so by synchronizing the various goroutines.